pinot-hadoop/src/main/java/com/linkedin/pinot/hadoop/job/mapper/HadoopSegmentCreationMapReduceJob.java

```diff
@@ -83,17 +81,27 @@ public void setup(Context context) throws IOException, InterruptedException {
       LOGGER.info("*********************************************************************");
       _properties = context.getConfiguration();
 
-      _outputPath = _properties.get("path.to.output");
-      _tableName = _properties.get("segment.table.name");
+      _outputPath = _properties.get(JobConfigConstants.PATH_TO_OUTPUT);
+      _tableName = _properties.get(JobConfigConstants.SEGMENT_TABLE_NAME);
       _postfix = _properties.get("segment.name.postfix", null);
 
       if (_outputPath == null || _tableName == null) {
         throw new RuntimeException(
             "Missing configs: " +
                 "\n\toutputPath: " +
-                _properties.get("path.to.output") +
+                _properties.get(JobConfigConstants.PATH_TO_OUTPUT) +
                 "\n\ttableName: " +
-                _properties.get("segment.table.name"));
+                _properties.get(JobConfigConstants.SEGMENT_TABLE_NAME));
+      }
+      String tableConfigString = _properties.get(JobConfigConstants.TABLE_CONFIG);
```

Comment:
Where do you run `properties.set(JobConfigConstants.TABLE_CONFIG, tableConfig)`? It should be somewhere in `SegmentCreationJob`